### PR TITLE
removed unused codes related to type_shapes and state 

### LINKED ISSUE #1554: Remove unused `type_shapes` and `state` writer code.
### Description

Remove code related to `connectGSDStateSignal`, `connectGSDShapeSpec`, `slotWriteGSDState`, `slotWriteGSDShapeSpec`, `restoreStateGSD`, `getWriteSignal()`, and all other associated signals and slots.

These functions are present in `GSDDumpWriter.h`, `GSDDumpWriter.cc`, `AnisoPotentialPair.h`, `AnisoPotentialPairDipole.h`, `PotentialPair.h`, and `IntegratorHPMCMono.h`.

### Motivation and context

This code is unused. It was replaced by `Logger` in HOOMD-blue v3.0.0.

### PR DESCRIPTION
**Pull request by melodyyzh**
## Description
Removed unused code in `GSDDumpWriter.h`, `AnisoPotentialPair.h`, `AnisoPotentialPairDipole.cc`, `PotentialPair.h`, and `IntegratorHPMCMono.h`. 
Removed functions including `connectGSDStateSignal`, `connectGSDShapeSpec`, `slotWriteGSDState`, `slotWriteGSDShapeSpec`, `restoreStateGSD`, `getWriteSignal`.

## Motivation and context

Resolves #1554 

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by melodyyzh** [comment]
I did not find relevant functions to remove in `GSDDumpWriter.cc`. Also, I think in issue #1554, `AnisoPotentialPairDipole.h` should be `AnisoPotentialPairDipole.cc`. Is it a typo?

**Comment by joaander** [comment]
Yes, there may be typos or omissions in #1554. Use grep or ripgrep to confirm that there are no remaining functions by those names. 
```
▶ rg GSDState
hoomd/GSDReader.cc
326:GSDStateReader::GSDStateReader(const std::string& name, const int64_t frame) : m_name(name)
374:GSDStateReader::~GSDStateReader()
379:std::vector<std::string> GSDStateReader::getAvailableChunks(const std::string& base)
398:pybind11::array GSDStateReader::readChunk(const std::string& name)
483:    pybind11::class_<GSDStateReader, std::shared_ptr<GSDStateReader>>(m, "GSDStateReader")
485:        .def("getAvailableChunks", &GSDStateReader::getAvailableChunks)
486:        .def("readChunk", &GSDStateReader::readChunk);

hoomd/GSDReader.h
107:    GSDStateReader provides an interface for ``from_state`` methods to discover and read state data
110:class PYBIND11_EXPORT GSDStateReader
118:    GSDStateReader(const std::string& name, const int64_t frame);
121:    ~GSDStateReader();
142:/// Exports GSDReader and GSDStateReader to python
(venv) cheme-walker:~/devel/hoomd ▶ rg GSDState
hoomd/GSDReader.cc
326:GSDStateReader::GSDStateReader(const std::string& name, const int64_t frame) : m_name(name)
374:GSDStateReader::~GSDStateReader()
379:std::vector<std::string> GSDStateReader::getAvailableChunks(const std::string& base)
398:pybind11::array GSDStateReader::readChunk(const std::string& name)
483:    pybind11::class_<GSDStateReader, std::shared_ptr<GSDStateReader>>(m, "GSDStateReader")
485:        .def("getAvailableChunks", &GSDStateReader::getAvailableChunks)
486:        .def("readChunk", &GSDStateReader::readChunk);

hoomd/GSDReader.h
107:    GSDStateReader provides an interface for ``from_state`` methods to discover and read state data
110:class PYBIND11_EXPORT GSDStateReader
118:    GSDStateReader(const std::string& name, const int64_t frame);
121:    ~GSDStateReader();
142:/// Exports GSDReader and GSDStateReader to python
```
```
▶ rg GSDShape
hoomd/CMakeLists.txt
154:    GSDShapeSpecWriter.h

hoomd/GSDShapeSpecWriter.h
17:class GSDShapeSpecWriter
20:    GSDShapeSpecWriter(const std::shared_ptr<const ExecutionConfiguration> exec_conf,

hoomd/md/AnisoPotentialPair.h
18:#include "hoomd/GSDShapeSpecWriter.h"

hoomd/md/PotentialPair.h
15:#include "hoomd/GSDShapeSpecWriter.h"

hoomd/hpmc/IntegratorHPMCMono.h
25:#include "hoomd/GSDShapeSpecWriter.h"
```
It looks like there is a `GSDShapeSpecWriter.h` that can go and a `GSDStateReader` class in `GSDReader.*`.

In `GSDDumpWriter.cc`, the unused code is `m_write_signal.emit(m_handle);` and correspondingly `hoomd::detail::SharedSignal<int(gsd_handle&)> m_write_signal;` in `GSDDumpWriter.h`.
```
▶ rg m_write_signal
hoomd/GSDDumpWriter.cc
401:    m_write_signal.emit(m_handle);

hoomd/GSDDumpWriter.h
224:    hoomd::detail::SharedSignal<int(gsd_handle&)> m_write_signal;
```

**Comment by melodyyzh** [comment]
pre-commit.ci autofix

**Comment by melodyyzh** [comment]
pre-commit.ci run

**Review by tommy-waltmann — APPROVED** [review]
Looks good! It's always good to remove unused code so things don't get unnecessarily cluttered.

File: hoomd/GSDShapeSpecWriter.h
Comment (joaander) [inline hoomd/GSDShapeSpecWriter.h:1]: Remove the whole file with `git rm hoomd/GSDShapeSpecWriter.h`.

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks! There are few more lines that can be removed, then this is good to merge.

File: hoomd/GSDShapeSpecWriter.h
Comment (melodyyzh) [reply hoomd/GSDShapeSpecWriter.h:1]: Got it, this file is now removed. I used rip grep to search for GSDState and GSDShape, and there are not any matching results, am I missing anything when you mentioned above that there are a few more lines to be removed?

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**